x-pack/plugins/kubernetes_security/public/components/tree_view_container/helpers.ts

```diff
@@ -28,20 +37,22 @@ export const addTreeNavSelectionToFilterQuery = (
       throw new Error('Invalid filter query');
     }
     parsedFilterQuery.bool.filter.push(
-      ...Object.keys(treeNavSelection).map((collectionKey) => {
-        const collection = collectionKey as KubernetesCollection;
-        return {
-          bool: {
-            should: [
-              {
-                match: {
-                  [KUBERNETES_COLLECTION_FIELDS[collection]]: treeNavSelection[collection],
+      ...Object.keys(treeNavSelection)
+        .filter((key) => key !== KubernetesCollection.clusterName)
+        .map((collectionKey) => {
+          const collection = collectionKey as KubernetesCollection;
```

Comment:
The fact that you needed to do this type of assertion shows that the current typing scheme is not so helpful, we might need to address some refactor on these types as part of technical debts

Reply:
I agree! We should definitely look into refactoring this along with other tech debts. I created an issue https://github.com/elastic/security-team/issues/4637. Feel free to add more to it :)

Reply:
well done with the issue, thanks!